### PR TITLE
Operator changes for process path collection in flowLogs

### PR DESCRIPTION
**Pull request by sridhartigera**
## Description
Process path and arguments are collected in flowLogs. In order to read the process path, arguments, felix will read /proc/pid/cmdline. This requires hostPID to be set to true in calico-node daemonset, set FELIX_FLOWLOGSCOLLECTPROCESSPATH env var. To enable/disable this feature, a new section (`collectProcessPath`) is added under LogCollector.

## For PR author

- [x] Tests for change.
- [x] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/logcollector/logcollector_controller.go
Comment (tmjd) [inline pkg/controller/logcollector/logcollector_controller.go:161]: I don't think we can do the fillDefaults in this function because we need the "raw" instance in the Reconcile to save as preDefaultPatchFrom. Otherwise the reconcile will not see the that we're adding the CollectProcessPath and NOT patch the resource with the default.

Comment (tmjd) [reply pkg/controller/logcollector/logcollector_controller.go:161]: I also think in the Reconcile we need to move the patching of the LogCollector resource earlier, especially now since the Core controller will be reading the resource.
Please also as this move I think you will need to move https://github.com/tigera/operator/pull/1394/files#diff-2b8c59da3e46730154a614a6d0ea55cfc27b17106a1e4bcc3fc8c12f794e8571R351-R358 the defaulting of  instance.Spec.AdditionalStores.Syslog.LogTypes into the fillDefaults function.
Hopefully this all makes sense, let me know if it doesn't.

**Review by tmjd — COMMENTED** [review]
Looking good. I think we need a final decision on if "privilegedFeatures" is needed in the API.
Also I think we should adjust the defaulting of LogCollector since the core controller depends on LogCollector now.

File: pkg/controller/logcollector/logcollector_controller.go
Comment (tmjd) [inline pkg/controller/logcollector/logcollector_controller.go:226]: Only the block here should have been moved into fillDefaults. The above retrieving of the managementClusterConnection and the validation that IDSEvents is not set should be left where it was in the Reconcile.

File: pkg/controller/logcollector/logcollector_controller.go
Comment (tmjd) [inline pkg/controller/logcollector/logcollector_controller.go:237]: I don't think we need this flag. In the core_controller we always patch so I think that is safe to do.

File: pkg/controller/logcollector/logcollector_controller.go
Comment (tmjd) [inline pkg/controller/logcollector/logcollector_controller.go:292]: Please move patching the instance to after this so we can Patch as soon as possible.

**Review by tmjd — CHANGES_REQUESTED** [review]

File: pkg/controller/logcollector/logcollector_controller.go
Comment (tmjd) [inline pkg/controller/logcollector/logcollector_controller.go:173]: Please add a comment on this function that includes what the slice returned contains.

File: pkg/controller/logcollector/logcollector_controller_test.go
Comment (tmjd) [inline pkg/controller/logcollector/logcollector_controller_test.go:139]: This test seems to be badly organized with almost everything in this "image reconciliation" context, but could you create a separate context to put this new `It` block in that is more appropriate. Maybe a new Context block of "defaulting logic".

**Review by tmjd — COMMENTED** [review]
Looking good. A couple of minor things, I think we're mainly waiting on final decision on the API and if we need the `privilegedFeatures` sub-section.

**Review by tmjd — COMMENTED** [review]
Has this been tested on a cluster that needs PodSecurityPolicies? https://kubernetes.io/docs/concepts/policy/pod-security-policy/#what-is-a-pod-security-policy suggests that access to hostPID is controlled with PodSecurityPolicy.

**Comment by sridhartigera** [comment]
> Has this been tested on a cluster that needs PodSecurityPolicies? https://kubernetes.io/docs/concepts/policy/pod-security-policy/#what-is-a-pod-security-policy suggests that access to hostPID is controlled with PodSecurityPolicy.

yes. Tested with pod-security-policy enabled and made changes to operator to fix the same.

File: pkg/render/node.go
Comment (tmjd) [inline pkg/render/node.go:1411]: I think this should be gated on `c.collectProcessPathEnabled()`.

**Review by tmjd — APPROVED** [review]
LGTM, definitely need to squash this one on merge :grinning:

**Pull request merged**
